From 2c0e87f7cc4515c8b8c3a067a651ed1e6d26ddab Mon Sep 17 00:00:00 2001
From: Kamil Lulko <kamil.lulko@gmail.com>
Date: Fri, 24 Jul 2026 21:42:00 +0200
Subject: [PATCH] support: handle shared libraries on uClinux

Readline's shared-object configuration only recognizes host OS names
beginning with "linux".  Buildroot's ARM FDPIC toolchain uses the
"uclinuxfdpiceabi" canonical OS name, so readline falls back to generic
shared-library settings.  That produces libreadline.so.8 instead of the
Linux-style libreadline.so.8.3 and leaves shlib-install without the
version components it needs to create linker symlinks.

Handle uClinux with the existing Linux shared-library settings and
permissions.

Upstream: not submitted
Signed-off-by: Kamil Lulko <kamil.lulko@gmail.com>
---
 support/shlib-install | 2 +-
 support/shobj-conf    | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/support/shlib-install b/support/shlib-install
index 3977cf23..bb373a1d 100755
--- a/support/shlib-install
+++ b/support/shlib-install
@@ -71,7 +71,7 @@ fi
 # Cygwin installs both a dll (which must go in $BINDIR) and an implicit
 # link library (in $libdir)
 case "$host_os" in
-hpux*|darwin*|macosx*|linux*|solaris2*)
+hpux*|darwin*|macosx*|linux*|uclinux*|solaris2*)
 	if [ -z "$uninstall" ]; then
 		chmod 755 ${INSTALLDIR}/${LIBNAME}
 	fi ;;
diff --git a/support/shobj-conf b/support/shobj-conf
index af60da27..15307a8a 100755
--- a/support/shobj-conf
+++ b/support/shobj-conf
@@ -123,7 +123,7 @@ sunos5*|solaris2*)
 	;;
 
 # All versions of Linux (including Gentoo/FreeBSD) or the semi-mythical GNU Hurd.
-linux*-*|gnu*-*|k*bsd*-gnu-*|midnightbsd*|freebsd*|dragonfly*)
+linux*-*|uclinux*-*|gnu*-*|k*bsd*-gnu-*|midnightbsd*|freebsd*|dragonfly*)
 	SHOBJ_CFLAGS=-fPIC
 	SHOBJ_LD='${CC}'
 	SHOBJ_LDFLAGS='-shared -Wl,-soname,$@'
-- 
2.50.1

